### PR TITLE
[SPARK-12304][STREAMING] Make Spark Streaming web UI display more fri…

### PR DESCRIPTION
**Pull request by lw-lin**
…endly Receiver graphs

Currently, the Spark Streaming web UI uses the same maxY when displays 'Input Rate Times& Histograms' and 'Per-Receiver Times& Histograms'. 

This may lead to somewhat un-friendly graphs: once we have tens of Receivers or more, every 'Per-Receiver Times' line almost hits the ground.

This issue proposes to calculate a new maxY against the original one, which is shared among all the `Per-Receiver Times& Histograms' graphs.

Before:
before-5

After:
after-5

### CODE REVIEW

**Comment by proflin** [comment]
@zsxwing could you take a look at this please? :-)

**Comment by zsxwing** [comment]
Jenkins, test this please

File: streaming/src/main/scala/org/apache/spark/streaming/ui/StreamingPage.scala [hunk 1]
```diff
@@ -392,8 +392,12 @@ private[ui] class StreamingPage(parent: StreamingTab)
       maxX: Long,
       minY: Double,
       maxY: Double): Seq[Node] = {
+    val maxYCalculated = listener.receivedEventRateWithBatchTime.map {
+      case (streamId, eventRates) => new EventRateUIData(eventRates).max.getOrElse(0.0)
```
Comment (srowen) [inline streaming/src/main/scala/org/apache/spark/streaming/ui/StreamingPage.scala]: This seems good @proflin. You'll need to rebase anyway, so I have one tiny suggestion: you can omit `streamId` here and write `_`. I suppose you can `flatMap` and move the `getOrElse` outside the map, but I don't think this matters.

**Comment by zsxwing** [comment]
retest this please

File: streaming/src/main/scala/org/apache/spark/streaming/ui/StreamingPage.scala (hunk 1 above)
Comment (andrewor14) [inline streaming/src/main/scala/org/apache/spark/streaming/ui/StreamingPage.scala]: actually you can probably just do
```
val maxYCalculated = listener.receivedEventRateWithBatchTime.values
  .map { case (_, eventRate) => eventRate }
  .reduceOption(math.max)
  .map(_.ceil.toLong)
  .getOrElse(0L)
```
is it the same? Then we don't have to create a new `EventRateUIData` every time. It's also more readable IMO.

**Comment by shaneknapp** [comment]
jenkins test this please

**Comment by shaneknapp** [comment]
jenkins test this please

File: streaming/src/main/scala/org/apache/spark/streaming/ui/StreamingPage.scala (hunk 1 above)
Comment (proflin) [inline streaming/src/main/scala/org/apache/spark/streaming/ui/StreamingPage.scala]: @srowen thank you for the suggestion. I'll rebase :-)

Comment (proflin) [inline streaming/src/main/scala/org/apache/spark/streaming/ui/StreamingPage.scala]: @andrewor14 I think your code is more readable too, so let's do it that way :-)

**Comment by proflin** [comment]
Closing this PR.
I'll start a new PR readdressing this issue as well as resolving the conflicts.

Thanks again for the reviews.

**Pull request closed without merging**